src/store/actions/onboarding.js

```diff
@@ -17,30 +17,51 @@ export const onboardingClient = () => async (dispatch, getState, lh) => {
     client.defaults.headers = {};
     const urlOnboard = "light_clients/";
     const onboardReq = await client.get(urlOnboard, { params: { address } });
-    // We get the data
+
+    // Get the onboarding data from the hub
     const {
+      transport_mode,
       display_name_to_sign,
       seed_retry,
       password_to_sign,
     } = onboardReq.data;
-    // Sign it
-    const signed_display_name = await resolver(display_name_to_sign, lh, true);
-    const signed_password = await resolver(password_to_sign, lh, true);
-    const signed_seed_retry = await resolver(seed_retry, lh, true);
-    // Prepare a body for the request with all the required data
-    const body = {
-      registration_data : {
-        address,
-        signed_password,
-        signed_display_name,
-        signed_seed_retry,
-        password: password_to_sign,
-        display_name: display_name_to_sign,
-        seed_retry,
-      },
-    };
+
+    // Check hub current transport implementation
+    let register_body = {};
+    if (transport_mode === "matrix") {
+      // Sign the onboarding matrix data
+      const signed_display_name = await resolver(
+        display_name_to_sign,
+        lh,
+        true
+      );
+      const signed_password = await resolver(password_to_sign, lh, true);
+      const signed_seed_retry = await resolver(seed_retry, lh, true);
+      // Prepare a body for the request with all the required data
+      register_body = {
+        registration_data: {
+          address,
+          signed_password,
+          signed_display_name,
+          signed_seed_retry,
+          password: password_to_sign,
+          display_name: display_name_to_sign,
+          seed_retry,
+        },
+      };
+    } else if (transport_mode === "rif-comms") {
```

Comment:
I think that maybe we can put the transport mode as an enum on javascript and also change rif-comms to rif-communications, i mean just because is transport mode that's all.

Reply:
agree with the enum, i will modify that.

regarding the literal "rif-comms" is exactly what the hub respond, so it must keep like that